### ISSUE TITLE
any_in, also in support

### ISSUE THREAD

**Issue by wtn** [joshuabowers/search_magic#8]
Partial matches would be useful.

**Comment by joshuabowers** [comment]
any_in/also_in support is a relatively simple alteration to the code, at least as far as invoking it to get the $in criteria rather than all_in to get the $all criteria is concerned. The broader concern is how to trigger this functionality: preferably, this would be done through some token which is passed in as part of the pattern which is being searched for. However, this would require a few more modifications to the (admittedly simple) way that the pattern string is converted into the database query.

A lot of search engines use a special token word -- e.g. "OR" -- to match items that contain either of two particular phrases. Unfortunately, going this route would require a greater amount of syntactical analysis of the pattern string than what is being done right now, and almost completely rules out the possibility of searching for the term "OR" without use of a selector:word phrase. Doing this, however, would allow for much more complex patterns: something like the following would be possible, if not a bit convoluted for non-power-users:

```
(foo:bar OR foo:baz) AND qux:10
```

On the other hand, a predecessor variant of search_magic that I wrote about a year ago introduced a special selector:word phrase which changed the way the search was performed, albeit for the entire search rather than for a sub-search. Whenever a pattern contained "mode:any", it used any_in; absence of this phrase, or the use of "mode:all", used all_in. This would be very easy to add in, and would provide a nice, convenient way to switch the functionality. Unfortunately, it is not very flexible nor very obvious. An example of what this would look like would be:

```
mode:any light bright night
mode:all light bright night
```

Do you have a preference for functionality? Potential use-cases that you would like to see?

**Comment by joshuabowers** [comment]
Provided a solution which follows the second suggestion provided in the previous post: a mode selector for switching between any and all. Available as of 25141f4e6f8fbf83f88d4c50d9017b763ad3bb7e.
